From cd173e05ef950a05abb2772125fd58fdc7911127 Mon Sep 17 00:00:00 2001
From: Hariprasad Kelam <hkelam@marvell.com>
Date: Tue, 20 Jul 2021 18:02:40 +0530
Subject: [PATCH 10/13] octeontx2-af: fix 50G ethtool mapping

commit f1400dce652fc46ff4f4fd350ffad88f6183f063 from
git@git.assembla.com:cavium/WindRiver.linux.git

On Octeontx2 silicon 50G links are mapped with two serdes lanes
and Octeontx3 silicon they are mapped with single serdes lane.

This patch updates the ethtool link mode values against cgx modes
accordingly. A Separate patch will be pushed to rename CGX_MODE enum.

Change-Id: Ieb1ecf84ebed218fb012755110729403dfbe8828
Signed-off-by: Hariprasad Kelam <hkelam@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/57001
Reviewed-by: Sunil Kovvuri Goutham <sgoutham@marvell.com>
Tested-by: Sunil Kovvuri Goutham <sgoutham@marvell.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/cgx.c  | 16 ++++++++--------
 .../marvell/octeontx2/nic/otx2_ethtool.c         | 16 ++++++++--------
 2 files changed, 16 insertions(+), 16 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/cgx.c b/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
index 2e3dc4a45f55..8b60369c0332 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
@@ -1136,19 +1136,19 @@ static void otx2_map_ethtool_link_modes(u64 bitmask,
 	case  ETHTOOL_LINK_MODE_10000baseKX4_Full_BIT:
 		set_mod_args(args, 40000, 0, 0, BIT_ULL(CGX_MODE_40GAUI_C2C));
 		break;
-	case  ETHTOOL_LINK_MODE_50000baseSR_Full_BIT:
+	case  ETHTOOL_LINK_MODE_50000baseSR2_Full_BIT:
 		set_mod_args(args, 50000, 0, 0, BIT_ULL(CGX_MODE_50G_C2C));
 		break;
 	case  ETHTOOL_LINK_MODE_56000baseKR4_Full_BIT:
 		set_mod_args(args, 50000, 0, 0, BIT_ULL(CGX_MODE_50G_4_C2C));
 		break;
-	case  ETHTOOL_LINK_MODE_50000baseLR_ER_FR_Full_BIT:
+	case  ETHTOOL_LINK_MODE_50000baseDR_Full_BIT:
 		set_mod_args(args, 50000, 0, 0, BIT_ULL(CGX_MODE_50G_C2M));
 		break;
-	case  ETHTOOL_LINK_MODE_50000baseCR_Full_BIT:
+	case  ETHTOOL_LINK_MODE_50000baseCR2_Full_BIT:
 		set_mod_args(args, 50000, 0, 1, BIT_ULL(CGX_MODE_50G_CR));
 		break;
-	case  ETHTOOL_LINK_MODE_50000baseKR_Full_BIT:
+	case  ETHTOOL_LINK_MODE_50000baseKR2_Full_BIT:
 		set_mod_args(args, 50000, 0, 1, BIT_ULL(CGX_MODE_50G_KR));
 		break;
 	case  ETHTOOL_LINK_MODE_10000baseLRM_Full_BIT:
@@ -1166,19 +1166,19 @@ static void otx2_map_ethtool_link_modes(u64 bitmask,
 	case  ETHTOOL_LINK_MODE_100000baseKR4_Full_BIT:
 		set_mod_args(args, 100000, 0, 1, BIT_ULL(CGX_MODE_100G_KR4));
 		break;
-	case  ETHTOOL_LINK_MODE_50000baseSR2_Full_BIT:
+	case  ETHTOOL_LINK_MODE_50000baseSR_Full_BIT:
 		set_mod_args(args, 50000, 0, 0,
 			     BIT_ULL(CGX_MODE_LAUI_2_C2C_BIT));
 		break;
-	case  ETHTOOL_LINK_MODE_50000baseDR_Full_BIT:
+	case  ETHTOOL_LINK_MODE_50000baseLR_ER_FR_Full_BIT:
 		set_mod_args(args, 50000, 0, 0,
 			     BIT_ULL(CGX_MODE_LAUI_2_C2M_BIT));
 		break;
-	case  ETHTOOL_LINK_MODE_50000baseCR2_Full_BIT:
+	case  ETHTOOL_LINK_MODE_50000baseCR_Full_BIT:
 		set_mod_args(args, 50000, 0, 1,
 			     BIT_ULL(CGX_MODE_50GBASE_CR2_C_BIT));
 		break;
-	case  ETHTOOL_LINK_MODE_50000baseKR2_Full_BIT:
+	case  ETHTOOL_LINK_MODE_50000baseKR_Full_BIT:
 		set_mod_args(args, 50000, 0, 1,
 			     BIT_ULL(CGX_MODE_50GBASE_KR2_C_BIT));
 		break;
diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c
index 967198f970ef..0c7e2d7bf69d 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c
@@ -1050,20 +1050,20 @@ static void otx2_get_link_mode_info(u64 link_mode_bmap,
 		ETHTOOL_LINK_MODE_40000baseCR4_Full_BIT,
 		ETHTOOL_LINK_MODE_40000baseKR4_Full_BIT,
 		ETHTOOL_LINK_MODE_10000baseKX4_Full_BIT,
-		ETHTOOL_LINK_MODE_50000baseSR_Full_BIT,
+		ETHTOOL_LINK_MODE_50000baseSR2_Full_BIT,
+		ETHTOOL_LINK_MODE_50000baseDR_Full_BIT,
 		ETHTOOL_LINK_MODE_56000baseKR4_Full_BIT,
-		ETHTOOL_LINK_MODE_50000baseLR_ER_FR_Full_BIT,
-		ETHTOOL_LINK_MODE_50000baseCR_Full_BIT,
-		ETHTOOL_LINK_MODE_50000baseKR_Full_BIT,
+		ETHTOOL_LINK_MODE_50000baseCR2_Full_BIT,
+		ETHTOOL_LINK_MODE_50000baseKR2_Full_BIT,
 		ETHTOOL_LINK_MODE_10000baseLRM_Full_BIT,
 		ETHTOOL_LINK_MODE_100000baseSR4_Full_BIT,
 		ETHTOOL_LINK_MODE_100000baseLR4_ER4_Full_BIT,
 		ETHTOOL_LINK_MODE_100000baseCR4_Full_BIT,
 		ETHTOOL_LINK_MODE_100000baseKR4_Full_BIT,
-		ETHTOOL_LINK_MODE_50000baseSR2_Full_BIT,
-		ETHTOOL_LINK_MODE_50000baseDR_Full_BIT,
-		ETHTOOL_LINK_MODE_50000baseCR2_Full_BIT,
-		ETHTOOL_LINK_MODE_50000baseKR2_Full_BIT,
+		ETHTOOL_LINK_MODE_50000baseSR_Full_BIT,
+		ETHTOOL_LINK_MODE_50000baseLR_ER_FR_Full_BIT,
+		ETHTOOL_LINK_MODE_50000baseCR_Full_BIT,
+		ETHTOOL_LINK_MODE_50000baseKR_Full_BIT,
 		ETHTOOL_LINK_MODE_100000baseSR2_Full_BIT,
 		ETHTOOL_LINK_MODE_100000baseLR2_ER2_FR2_Full_BIT,
 		ETHTOOL_LINK_MODE_100000baseCR2_Full_BIT,
-- 
2.31.1

